modules/search/instant-search/store/effects.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Internal dependencies
+ */
+import { search } from '../lib/api';
+import { SORT_DIRECTION_ASC, VALID_SORT_KEYS } from '../lib/constants';
+import { getFilterKeys } from '../lib/filters';
+import { getQuery, setQuery } from '../lib/query-string';
+import {
+	recordFailedSearchRequest,
+	recordSuccessfulSearchRequest,
+	setFilter,
+	setSearchQuery,
+	setSort,
+} from './actions';
+
+/**
+ * Effect handler which will fetch search results from the API.
+ *
+ * @param {object} action - Action which had initiated the effect handler.
+ * @param {object} store -  Store instance.
+ */
+function makeSearchAPIRequest( action, store ) {
+	search( action.options )
+		.then( response => {
+			if ( response === null ) {
+				// Request has been cancelled by a more recent request.
+				return;
+			}
+
+			store.dispatch( recordSuccessfulSearchRequest( { options: action.options, response } ) );
+		} )
+		.catch( error => {
+			// eslint-disable-next-line no-console
+			console.error( 'Jetpack Search encountered an error:', error );
+			store.dispatch( recordFailedSearchRequest( error ) );
+		} );
+}
+
+function initializeQueryValues( action, store ) {
+	const queryObject = getQuery();
+
+	//
+	// Initialize search query value for the reducer.
+	//
+	store.dispatch( setSearchQuery( queryObject.s, false ) );
+
+	//
+	// Initialize sort value for the reducer.
+	//
+	let sort = 'revelance';
```

Comment:
revelance -> relevance.

Should we perhaps have a DEFAULT_SORT constant?

Reply:
Fixed! Added `DEFAULT_SORT_KEY` constant.